docs/cli.md

```diff
@@ -287,10 +287,11 @@ If you just want to update a few packages and not all, you can list them as such
 poetry update requests toml
 ```
 
-Note that this will not update versions for dependencies outside their version constraints specified
-in the `pyproject.toml` file. In other terms, `poetry update foo` will be a no-op if the version constraint
-specified for `foo` is `~2.3` or `2.3` and `2.4` is available. In order for `foo` to be updated, you must
-update the constraint, for example `^2.3`. You can do this using the `add` command.
+Note that this will not update versions for dependencies outside their (SemVer) version constraints specified
```

Comment:
most of the additions here are unhelpful for people who have written their constraints not using the caret.  

Reproducing a tiny fragment of https://python-poetry.org/docs/dependency-specification/#version-constraints here is redundant - and too hyper-focused on the particular misunderstanding that you happened to have.

Suggest that simply cross-referencing from the words "version constraints" to https://python-poetry.org/docs/dependency-specification/#version-constraints should be sufficient

Reply:
Done!

Reply:
the addition of "(Semver)" is still unwanted: only caret constraints are semver constraints

what I had in mind was simply 

> Note that this will not update versions for dependencies outside their [version constraints]({{ cross-reference here}}) specified ...

where an internal cross-reference is the right way to do that rather than pointing directly at  python-poetry.org (but I'm not familiar with the exact syntax needed so you'll have either to figure it out yourself or take a couple of tries to get it right, sorry!)